### PR TITLE
Adds Voice Activity Detection, Emotion Detection, and Question Answering Nodes

### PR DESCRIPTION
**Pull request by derkmed**

### CODE REVIEW

File: ros/angel_system_nodes/angel_system_nodes/emotion_detector.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    def _apply_filter(self, msg):
+        '''
+        Abstracts away any filtering to apply on received messages. Return
+        none if the message should be filtered out. Else, return the incoming
+        msg if it can be included.
+        '''
+        # if msg.user_intent.lower() == "user inquiry":
+        #     return msg
+        # else:
+        #     return None
```
Comment (derkmed) [inline ros/angel_system_nodes/angel_system_nodes/emotion_detector.py:176]: Currently no filtering is applied.

File: docker/workspace-build/Dockerfile [hunk 2]
```diff
@@ -90,7 +90,7 @@ RUN apt-get -y update \
 COPY poetry.lock pyproject.toml docker/workspace_build_pydep_install.sh "${ANGEL_WORKSPACE_DIR}"/
 RUN cd "${ANGEL_WORKSPACE_DIR}" \
  && "${ANGEL_WORKSPACE_DIR}/workspace_build_pydep_install.sh"
-RUN python -m nltk.downloader punkt
+# RUN python -m nltk.downloader punkt
```
Comment (derkmed) [inline docker/workspace-build/Dockerfile:93]: During the demo with @ovenmitt , we found this to be insufficient for whatever reason.

**Review by derkmed — COMMENTED** [review]
Adds the latest VAD, Emotion Detection (Heuristically driven and GPT-prompted versions), Intent Detection (Heuristically driven and GPT-prompted versions), and Question-Answer (GPT-prompted)  nodes.

**Comment by hdefazio** [comment]
@derkmed Can this MR be closed in favor of  your new ones?

**Comment by derkmed** [comment]
@hdefazio We can deprecate this PR once 
- https://github.com/PTG-Kitware/angel_system/pull/295
- https://github.com/PTG-Kitware/angel_system/pull/296
- https://github.com/PTG-Kitware/angel_system/pull/297
- https://github.com/PTG-Kitware/angel_system/pull/298
- https://github.com/PTG-Kitware/angel_system/pull/299
all get merged.

I believe those *should* contain all the necessary changes.

**Pull request closed without merging**